src/gui/gui.cpp

```diff
@@ -51,10 +51,32 @@ void init()
 	try {
 		schema_validation::schema_validator validator(filesystem::get_wml_location("schema/gui.cfg"));
 
+		// Core theme files
 		preproc_map preproc(game_config::config_cache::instance().get_preproc_map());
 		filesystem::scoped_istream stream = preprocess_file(filesystem::get_wml_location("gui/_main.cfg"), &preproc);
-
 		read(cfg, *stream, &validator);
+
+		// Add-on theme files
+		// Append the $user_campaign_dir/*/gui.cfg files to the main gui config.
+		std::vector<std::string> user_dirs;
+		{
+			const std::string user_campaign_dir = filesystem::get_addons_dir();
+			std::vector<std::string> user_files;
+			filesystem::get_files_in_dir(
+				user_campaign_dir, &user_files, &user_dirs, filesystem::name_mode::ENTIRE_FILE_PATH);
+		}
+
+		game_config::config_cache& cache(game_config::config_cache::instance());
+		for(const std::string& umc : user_dirs) {
+			const std::string gui_file = umc + "/gui-theme.cfg";
+			if(filesystem::file_exists(gui_file)) {
+				config gui_cfg;
+				cache.get_config(gui_file, gui_cfg);
+				validator.validate(gui_cfg, "gui", 0, gui_file);
+				cfg.append(gui_cfg);
+			}
+		}
+
```

Comment:
Comment and error don’t fit anymore.

Reply:
are the changes in 3c074a027c21b434e8834388e065779ac5903a1a adequate?

Reply:
Using `config_cache` seemed like a good idea.